ipv8/test/base.py

```diff
@@ -144,6 +144,4 @@ def introduce_nodes(self):
         yield self.deliver_messages()
 
     def temporary_directory(self):
-        d = os.path.join("temp", self.__class__.__name__ + str(int(time.time())))
-        os.makedirs(d)
-        return d
+        return mkdtemp()
```

Comment:
This is the root cause of a lot of problems, you do not have permissions on the temp folder of a lot of operating systems. `mkdtemp` is horribly broken and should never be used anywhere.

Reply:
Previous version is the only one that actually works on Windows.

Reply:
It fails on Linux, though, when run in parallel. Because it deletes the whole "temp" subtree, that could be used by other tests at that moment. Also, the combination of class and timestamp are not random enough and could result in subtle errors when tests run in parallel.